### PR TITLE
React Hooks Rewrite

### PR DESCRIPTION
**Pull request by archcorsair**
Rewrote the component using hooks instead

@steelbrain please don't kill me I removed your presets xD

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-  "presets": ["steelbrain"]
+  "presets": ["@babel/preset-env"]
```
Comment (steelbrain) [inline .babelrc:2]: You did not specify **which** env, specify some `targets` for it please

File: package.json [hunk 2]
```diff
@@ -3,11 +3,14 @@
   "version": "1.0.3",
   "description": "Load scripts with ease",
   "main": "lib/index.js",
+  "typings": "lib/typings/index.d.ts",
```
Comment (steelbrain) [inline package.json:6]: Missing `module` key I think

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ node_modules
 /lib
 *.DS_Store
 spec/fixtures/ignored
+.history
```
Comment (steelbrain) [inline .gitignore:6]: Missing EOL

File: package.json (hunk 2 above)
Comment (steelbrain) [reply package.json:6]: Bump

File: src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    const scripts: HTMLScriptElement[] = Array.from(document.querySelectorAll('script'))
+    if (scripts.find((script) => script.src === src)) {
+      onLoad()
+      return
+    }
+    const script: HTMLScriptElement = document.createElement('script')
+    script.src = src
+    script.onload = onLoad
+    document.body.appendChild(script)
+  }, [src, onLoad])
```
Comment (steelbrain) [inline src/index.tsx:19]: Remove the dependency on `src, onLoad`. You only want this to fire once, ever.

Even in the old code, `componentDidMount` was only triggered once

File: package.json (hunk 2 above)
Comment (archcorsair) [reply package.json:6]: This was fixed in last push already

File: src/index.tsx (hunk 4 above)
Comment (archcorsair) [reply src/index.tsx:19]: Should I override the eslint rule for this? I did it based on this comment within this discussion:
https://github.com/facebook/react/issues/14920#issuecomment-470913287

Comment (steelbrain) [reply src/index.tsx:19]: Yes you should override for this specific case

**Pull request merged**